### PR TITLE
feat: add remaining instance policies

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
adds instance policies available in the provider
remove deprecated RestAPI provider

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Features:
- added the ability to configure rotation policy using new vars `rotation_enabled` (default value of true), and `rotation_interval_month` (default value of 1)
- added the ability to enable / disable a dual authorization policy on the instance using new var `dual_auth_delete_enabled` (default value of false). Note: Once the dual authorization policy is set on the instance, it cannot be reverted. An instance with dual authorization policy enabled cannot be destroyed using Terraform. 
- added the ability to configure key create import access policy settings using vars `key_create_import_access_enabled` (default value of true) and `key_create_import_access_settings` (If no value is passed it takes all of the objects default values. If only one is passed then it only overrides that one value so consumers do not have to pass the entire map in order to supply overrides they want to use).

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by ocofaigh** [comment]
@MatthewLemmond can you resolve conflicts please?

File: tests/pr_test.go [hunk 1]
```diff
@@ -12,12 +12,19 @@ import (
 const resourceGroup = "geretain-test-key-protect"
 const terraformDir = "examples/default"
 
+var ignoreUpdates = []string{
+	"module.key_protect_module.ibm_kms_instance_policies.key_protect_instance_policies",
```
Comment (ocofaigh) [inline tests/pr_test.go:16]: Is this just a once off for this PR? If so, please leave a code comment here stating that it should be removed once this PR is merged

File: variables.tf [hunk 2]
```diff
@@ -45,12 +45,53 @@ variable "plan" {
...
+variable "key_create_import_access_settings" {
+  type = object({
+    create_root_key     = optional(bool, true)
+    create_standard_key = optional(bool, true)
+    import_root_key     = optional(bool, true)
+    import_standard_key = optional(bool, true)
+    enforce_token       = optional(bool, false)
+  })
+  description = "Key create import access policy settings to configure if var.enable_key_create_import_access_policy is true. For more info see https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-keyCreateImportAccess"
+  default     = {}
```
Comment (ocofaigh) [inline variables.tf:92]: @MatthewLemmond I'm just finalising the release notes for this PR to be merged, so I need to know what this default value of empty map means. Does it mean they get the default values in the object?

Comment (MatthewLemmond) [reply variables.tf:92]: Correct, if no value is passed it takes all of the default values, if only one is passed then it only overrides that one value so consumers do not have to pass the entire map in order to supply overrides they want to use.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**